Kubeadm e2e

liztio opened the pull request:
**What this PR does / why we need it**:

Provides in-tree E2E tests for the Kubeadm subproject

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes [kubeadm#456](https://github.com/kubernetes/kubeadm/issues/456)

**Special notes for your reviewer**:

The weird way tests are executed mirrors `e2e_node`. A future pull request will add a frontend for these tests to kubetest, which will abstract away much of this detail.

**Release note**:
```release-note
NONE
```

liztio commented:
/unset wip

timothysc commented on test/e2e_kubeadm/runner/local/run_local.go:
```diff
@@ -0,0 +1,106 @@
+var testFlags = flag.String("test-flags", "", "Space-separated list of arguments to pass to node e2e test.")
+var build = flag.Bool("build", false, "use Bazel to build binaries before testing")
+
+func main() {
```
why is this here?

timothysc commented on test/e2e_node/builder/build.go:
```diff
@@ -38,18 +38,18 @@ var buildTargets = []string{
 func BuildGo() error {
-	glog.Infof("Building k8s binaries...")
-	k8sRoot, err := GetK8sRootDir()
+	glog.Info("Building k8s binaries...")
```
This seems like an orthogonal set of changes which should be their own PR.

timothysc reviewed (changes requested):
IMO this PR should be split into multiple.

liztio committed:
Generated artefacts

liztio committed:
Basic E2E tests for kubeadm

Run with `go run test/e2e_kubeadm/runner/local/run_local.go -build -test-flags="--kubeconfig=$KUBECONFIG"`

liztio commented:
@timothysc the utils PR has been merged, this PR now only the E2E-specific changes

timothysc commented on test/e2e_kubeadm/runner/local/run_local.go:
```diff
@@ -0,0 +1,106 @@
+	"k8s.io/kubernetes/test/utils"
+)
+
+func bazelBuild() error {
```
I'm still confused why this is needed.

liztio replied:
it's just how the e2e_node tests work. The local runner is responsible for compiling the artefact. I could take it out, but that responsibility has to be somewhere.

timothysc reviewed (commented):
Why do you need the bazel shim, does the bazel build not properly create the test artifact?

timothysc reviewed (approved):
/lgtm 
/approve 

I'll circle back on the build mechanics later.

timothysc commented:
/retest
/lgtm

timothysc commented:
/retest

timothysc commented:
/ok-to-test

timothysc commented:
/retest

liztio commented:
/test pull-kubernetes-bazel-test

timothysc commented:
/lgtm
